src/Api/Models/User/User.cs

```diff
@@ -65,5 +65,8 @@ public bool IsStudent()
         {
             return Role == BOL.Models.Role.Student;
         }
+
+        [ForeignKey("Id")]
+        public virtual List<Notification> NotificationsForUser { get; set; }
```

Comment:
'Notifications' will be a better name, because it only contains the Notifications related to the current User/Instance.
Id is may not a clearly name for a identifier. Because which id? We have three in this class.